### PR TITLE
Fix html referencing wrong bundle

### LINKED ISSUE #4772: Referencing a css file from js and html builds incorrect html script references
# 🐛 bug report

When you build an app that has multiple entrypoints which both point to the same JS and CSS file (this is a minimal repro use-case) and also have that JS file point to the same CSS file the html bundle that gets built returns the css bundle path in 2 `link` tags, the one that was already present and a new one inserted as it's also referenced by the JS. Besides this is also puts that css bundle path in the script tag, where the JS bundle should be referenced.

## 🤔 Expected Behavior

The script tag should reference the JS bundle and ideally but optionally the CSS bundle should only be referenced once. So basically not add an additional link tag if the css file is already part of the html bundle.

## 😯 Current Behavior

JS does not get loaded and CSS bundle is referenced 3 times.

## 💁 Possible Solution

I solved this by just removing the css reference in the JS file, which is a good solution for this specific use-case as it should really only be referenced once.

But this seems like a bug that could potentially have a larger impact on more useful use-cases.

Tried to debug it as far as I could and got stuck at `replaceURLReferences` where `bundleGraph.getBundlesInBundleGroup(resolved.value)` returned 3 bundles with the first being the JS which I wanted to reference in the script tag.

And the css bundle which is both bundle 1 and 2 in the link tag. Maybe this is a bundler bug as these are all the same `bundle-group`? and it should probably be 2 groups, one for the CSS and one for the JS. I wish I knew more about the graph and bundling logic of Parcel but it's currently a bit of a black box so I should probably focus my efforts towards the parts I know a lot about,

## 💻 Code Sample

Reproducible example: (also made a branch with a test-case: https://github.com/parcel-bundler/parcel/tree/fix-html-bug)

`parcel build ./*.html`

`a.html`
```html
<head>
	<link rel="stylesheet" href="./style.css">
</head>

<body>
	<script src="./index.js"></script>
</body>
```

`b.html`
```html
<head>
	<link rel="stylesheet" href="./style.css">
</head>

<body>
	<script src="./index.js"></script>
</body>
```

`index.js`
```js
import "./style.css";
```

`index.css`
```css
html, body {}
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | v2.0.0-beta.1

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

This PR fixes a bug in `replaceURLReferences` where it sometimes returns a reference to an incorrect bundle as it does not actually use the entryBundle.

Thanks @mischnic for mentioning bundle_group has an entryAssetId

See #4772 for context

Fixes #4772

### CODE REVIEW

File: packages/core/utils/src/replaceBundleReferences.js [hunk 1]
```diff
@@ -64,8 +64,13 @@ export function replaceURLReferences({
     }
 
     invariant(resolved.type === 'bundle_group');
-    let entryBundle = bundleGraph.getBundlesInBundleGroup(resolved.value).pop();
-    if (entryBundle.isInline) {
+    let entryAssetId = resolved.value.entryAssetId;
+    let entryBundle = bundleGraph
+      .getBundlesInBundleGroup(resolved.value)
+      .find(b => {
+        return !!b.getEntryAssets().find(a => a.id === entryAssetId);
```
Comment (DeMoorJasper) [inline packages/core/utils/src/replaceBundleReferences.js:71]: Maybe we should expose the entryAssetIds to speed this up a bit?

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.34s | -132.00ms |
| Cached | 2.27s | +14.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 232.00ms | -13.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 233.00ms | -12.00ms 🚀 |
| dist/legacy/kitchen-sink.679578a5.js | 1.16kb | +0.00b | 1.68s | -102.00ms 🚀 |
| dist/modern/kitchen-sink.90de237b.js | 1.16kb | +0.00b | 1.68s | -101.00ms 🚀 |
| dist/legacy/index.html | 709.00b | +0.00b | 646.00ms | -86.00ms 🚀 |
| dist/modern/index.html | 709.00b | +0.00b | 644.00ms | -87.00ms 🚀 |
| dist/legacy/styles.afb8e31a.css | 78.00b | +0.00b | 1.68s | -101.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 235.00ms | +53.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 234.00ms | +53.00ms ⚠️ |
| dist/legacy/kitchen-sink.679578a5.js | 1.16kb | +0.00b | 494.00ms | +79.00ms ⚠️ |
| dist/modern/kitchen-sink.90de237b.js | 1.16kb | +0.00b | 493.00ms | +78.00ms ⚠️ |
| dist/legacy/index.html | 709.00b | +0.00b | 496.00ms | +79.00ms ⚠️ |
| dist/modern/index.html | 709.00b | +0.00b | 493.00ms | +78.00ms ⚠️ |
| dist/legacy/styles.afb8e31a.css | 78.00b | +0.00b | 494.00ms | +79.00ms ⚠️ |
| dist/modern/styles.cda8c91b.css | 78.00b | +0.00b | 493.00ms | +78.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 28.58s | +973.00ms |
| Cached | 2.44s | +49.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 478.42kb | +0.00b | 302.00ms | -68.00ms 🚀 |
| dist/PermalinkedComment.89fe2d7f.js | 4.61kb | +0.00b | 300.00ms | -63.00ms 🚀 |
| dist/UserProfile.8cce6c4b.js | 1.83kb | +0.00b | 299.00ms | -64.00ms 🚀 |
| dist/NotFound.98e75551.js | 468.00b | +0.00b | 299.00ms | -64.00ms 🚀 |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 208.00ms | -47.00ms 🚀 |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.95m | -4.22s |
| Cached | 5.43s | +334.00ms ⚠️ |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/pdfRenderer.74642224.js | 1.11mb | +0.00b | 54.91s | -11.49s 🚀 |
| dist/editorView.918fe31c.js | 586.93kb | +0.00b | 1.45m | -6.64s 🚀 |
| dist/popup.fe67861a.js | 218.11kb | +0.00b | 58.10s | -8.23s 🚀 |
| dist/EmojiPickerComponent.20f6055b.js | 98.66kb | +0.00b | 14.33s | -45.62s 🚀 |
| dist/Toolbar.a68787f4.js | 93.03kb | +0.00b | 1.01m | -7.91s 🚀 |
| dist/card.e4d8099d.js | 61.60kb | +0.00b | 16.02s | +1.04s ⚠️ |
| dist/component.df0df6bb.js | 56.73kb | +0.00b | 7.45s | -1.08s 🚀 |
| dist/Modal.c635528d.js | 40.56kb | +0.00b | 4.07s | -402.00ms 🚀 |
| dist/EmojiPickerComponent.38775e63.js | 18.33kb | +0.00b | 11.98s | -45.47s 🚀 |
| dist/js.baeda348.js | 16.46kb | +0.00b | 7.35s | -1.02s 🚀 |
| dist/browser.24942986.js | 16.27kb | +0.00b | 59.52s | -8.03s 🚀 |
| dist/dropzone.35a61bda.js | 14.89kb | +0.00b | 59.52s | -43.34s 🚀 |
| dist/ui.9e4d74b0.js | 14.45kb | +0.00b | 15.94s | +960.00ms ⚠️ |
| dist/workerHasher.ebf0b386.js | 11.83kb | +0.00b | 1.66m | +1.43m ⚠️ |
| dist/EmojiTypeAheadComponent.899f7d25.js | 9.20kb | +0.00b | 55.44s | +43.62s ⚠️ |
| dist/EmojiPickerComponent.dc1e6b3c.js | 3.92kb | +0.00b | 11.98s | -47.30s 🚀 |
| dist/16.cb2f1a3a.js | 3.72kb | +0.00b | 4.06s | -410.00ms 🚀 |
| dist/Modal.1111ef83.js | 3.50kb | +0.00b | 4.06s | -3.70s 🚀 |
| dist/png-chunks-extract.c0f57e67.js | 3.49kb | +0.00b | 7.35s | -1.02s 🚀 |
| dist/ak-editor.e6be06da.css | 3.46kb | +0.00b | 1.23m | -6.49s 🚀 |
| dist/dropzone.79e05ece.js | 3.44kb | +0.00b | 54.91s | +14.39s ⚠️ |
| dist/clipboard.f1852e5e.js | 2.98kb | +0.00b | 59.20s | -8.35s 🚀 |
| dist/16.c7be7376.js | 2.54kb | +0.00b | 7.35s | -1.02s 🚀 |
| dist/EmojiUploadComponent.4946a1c6.js | 2.50kb | +0.00b | 55.44s | +43.61s ⚠️ |
| dist/card.444c6665.js | 2.19kb | +0.00b | 30.62s | -9.75s 🚀 |
| dist/16.bb063a1f.js | 1.83kb | +0.00b | 7.75s | -1.46s 🚀 |
| dist/media-picker-analytics-error-boundary.187d26a0.js | 1.82kb | +0.00b | 58.00s | -9.35s 🚀 |
| dist/date.56d1aa9e.js | 1.81kb | +0.00b | 7.98s | -1.47s 🚀 |
| dist/16.b1b0372d.js | 1.77kb | +0.00b | 7.55s | -2.56s 🚀 |
| dist/images.16b3af40.js | 1.75kb | +0.00b | 8.06s | -1.51s 🚀 |
| dist/feedback.b59078b6.js | 1.72kb | +0.00b | 8.79s | -1.67s 🚀 |
| dist/workerHasher.9975765a.js | 1.65kb | +0.00b | 55.44s | +43.62s ⚠️ |
| dist/list-number.d2097959.js | 1.54kb | +0.00b | 8.18s | -1.55s 🚀 |
| dist/status.f61d9ac5.js | 1.54kb | +0.00b | 8.36s | -1.56s 🚀 |
| dist/16.450f3a0d.js | 1.49kb | +0.00b | 7.75s | -935.00ms 🚀 |
| dist/code.4289d7dd.js | 1.46kb | +0.00b | 7.94s | -1.47s 🚀 |
| dist/16.2b801dd1.js | 1.44kb | +0.00b | 7.55s | -2.56s 🚀 |
| dist/16.fe51c023.js | 1.44kb | +0.00b | 7.54s | -1.14s 🚀 |
| dist/16.7d4b2402.js | 1.42kb | +0.00b | 7.94s | -1.47s 🚀 |
| dist/link.36b44af7.js | 1.38kb | +0.00b | 8.18s | -1.55s 🚀 |
| dist/heading6.675c185b.js | 1.38kb | +0.00b | 8.79s | -1.67s 🚀 |
| dist/heading3.4e88bcd6.js | 1.37kb | +0.00b | 8.40s | -1.56s 🚀 |
| dist/16.e32aa49c.js | 1.34kb | +0.00b | 4.06s | -410.00ms 🚀 |
| dist/emoji.3654aec7.js | 1.31kb | +0.00b | 8.06s | -1.51s 🚀 |
| dist/media-card-analytics-error-boundary.849f68c6.js | 1.31kb | +0.00b | 33.33s | -7.19s 🚀 |
| dist/16.d1b58d7c.js | 1.29kb | +0.00b | 7.94s | -1.47s 🚀 |
| dist/16.1dbb5df4.js | 1.28kb | +0.00b | 4.06s | -410.00ms 🚀 |
| dist/Modal.6509cbe3.js | 1.26kb | +0.00b | 7.35s | -1.02s 🚀 |
| dist/heading5.e338f1e8.js | 1.25kb | +0.00b | 8.79s | -1.67s 🚀 |
| dist/16.ceac57cd.js | 1.25kb | +0.00b | 7.75s | -1.21s 🚀 |
| dist/expand.4d883914.js | 1.24kb | +0.00b | 8.79s | -1.67s 🚀 |
| dist/16.0efadc10.js | 1.23kb | +0.00b | 7.55s | -2.63s 🚀 |
| dist/16.35be54a4.js | 1.23kb | +0.00b | 7.60s | -1.08s 🚀 |
| dist/16.6be9bbc8.js | 1.23kb | +0.00b | 7.75s | -941.00ms 🚀 |
| dist/16.9015e8cf.js | 1.23kb | +0.00b | 8.24s | -1.49s 🚀 |
| dist/16.234ee133.js | 1.23kb | +0.00b | 7.61s | +3.14s ⚠️ |
| dist/heading2.89a8e126.js | 1.19kb | +0.00b | 8.40s | -1.56s 🚀 |
| dist/16.7ca41ffe.js | 1.16kb | +0.00b | 7.86s | -1.34s 🚀 |
| dist/mention.ae379620.js | 1.15kb | +0.00b | 8.18s | -1.55s 🚀 |
| dist/heading4.b14d5ea4.js | 1.14kb | +0.00b | 8.46s | -1.61s 🚀 |
| dist/layout.d9df99ef.js | 1.13kb | +0.00b | 8.10s | -1.51s 🚀 |
| dist/divider.4476c3de.js | 1.11kb | +0.00b | 8.06s | -1.51s 🚀 |
| dist/quote.4733acb8.js | 1.10kb | +0.00b | 10.93s | +1.10s ⚠️ |
| dist/action.b9409056.js | 1.08kb | +0.00b | 7.94s | -1.47s 🚀 |
| dist/decision.77ee0aaa.js | 1.07kb | +0.00b | 8.06s | -1.51s 🚀 |
| dist/panel-warning.e55309d2.js | 1.06kb | +0.00b | 8.27s | -1.56s 🚀 |
| dist/list.cf500a90.js | 1.03kb | +0.00b | 8.18s | -1.55s 🚀 |
| dist/heading1.165705a7.js | 1.03kb | +0.00b | 8.40s | -1.56s 🚀 |
| dist/media-card-analytics-error-boundary.b3d7f02c.js | 995.00b | +0.00b | 15.94s | +936.00ms ⚠️ |
| dist/panel-error.364fbf3c.js | 987.00b | +0.00b | 8.23s | -1.53s 🚀 |
| dist/panel.e4fe05d1.js | 978.00b | +0.00b | 8.27s | -1.56s 🚀 |
| dist/table.fd1280b6.js | 970.00b | +0.00b | 8.40s | -1.56s 🚀 |
| dist/panel-success.66723ab0.js | 926.00b | +0.00b | 8.27s | -1.56s 🚀 |
| dist/panel-note.a8b696c4.js | 922.00b | +0.00b | 8.27s | -48.51s 🚀 |
| dist/index.html | 127.00b | +0.00b | 6.22s | -400.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/ak-editor.4341ac4e.js | 2.69mb | -6.00b 🚀 | 608.00ms | +16.00ms |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/268942862590149133)**

File: packages/core/utils/src/replaceBundleReferences.js (hunk 1 above)
Comment (mischnic) [reply packages/core/utils/src/replaceBundleReferences.js:71]: Should it be `b.getMainEntry()?.id === entryAssetId` anyway? Not sure

Comment (DeMoorJasper) [reply packages/core/utils/src/replaceBundleReferences.js:71]: Not sure if it's multiple entries it might result in undefined?

**Pull request merged**